convert2rhel/subscription.py

```diff
@@ -413,3 +415,44 @@ def check_needed_repos_availability(repo_ids_needed):
             all_repos_avail = False
     if all_repos_avail:
         loggerinst.info("Needed RHEL repos are available.")
+
+
+def download_subscription_manager_packages():
+    loggerinst = logging.getLogger(__name__)
+
+    _CENTOS_CONTRIB_REPO = '[contrib]\n' \
+                           'name=CentOS-$releasever - Contrib\n' \
+                           'mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=contrib\n' \
+                           'gpgcheck=0\n' \
+                           'enabled=1\n' \
+                           'protect=0\n'
+
+    _TEMP_REPOFILE_DIR = '/tmp/convert2rhel/centos_contrib.repo'
+
+    version = utils.parse_version_number(system_info.system_release_file_content)
+
+    if not version:
+        loggerinst.warning(msg='Unable to detect system version')
+        sys.exit(1)
+
+    if not version:
+        loggerinst.warning(msg='Unable to detect system version')
+        sys.exit(1)
+
+    if int(version['major']) <= 6 and ('minor' not in version or ('minor' in version and int(version['minor']) <= 6)):
+        loggerinst.error(msg='Unable to download subscription-manager on CentOS/Oracle Linux 6.6 and older. '
+                             'Please use the --disable-submgr option or update to CentOS/Oracle Linux 6.7 or later')
+        sys.exit(1)
+
+    if int(version['major']) == 6:
+        utils.store_content_to_file(filename=_TEMP_REPOFILE_DIR, content=_CENTOS_CONTRIB_REPO)
+        pkghandler.call_yum_cmd(command="install",
+                                args="subscription-manager subscription-manager-rhsm subscription-manager-rhsm-certificates"
+                                     " --setopt=reposdir=/tmp/convert2rhel/ --downloadonly"
+                                     " --downloaddir=/usr/share/subscription-manager/")
+
+    elif int(version['major']) == 7:
+        pkghandler.call_yum_cmd(command="install", args="subscription-manager subscription-manager-rhsm "
+                                                        "subscription-manager-rhsm-certificates "
+                                                        "--downloadonly --downloaddir=/usr/share/subscription-manager/")
+        pkghandler.call_yum_cmd(command="install", args="python-rhsm-certificates")
```

Comment:
If they use a system we don't support, e.g. major version 5, we'll get no logs at all. Add a log at the end of the function in an else-branch for when we don't support a system.